lib/matplotlib/backend_bases.py

```diff
@@ -138,6 +145,35 @@ def get_registered_canvas_class(format):
     return backend_class
 
 
+class MainLoopBase(object):
```

Comment:
Might as well make this an ABC so that we can make `begin` and `end` abstract methods.